src/ShopBundle/objects/WebModules/ViewMyOrderDetails/MTShopViewMyOrderDetails.php

```diff
@@ -85,24 +85,39 @@ public function Accept(
             'order' => null,
         );
 
-        if (false === $viewOrderDetailHandler->orderIdBelongsToUser($orderIdRequested, $extranetUserId)) {
-            // this is not our order. access denied
-            $viewData['error'] = true;
-            $viewData['errorCode'] = 'notMyOrderError';
-        }
+        $order = $this->getOrder($orderIdRequested);
 
-        $order = $this->getDbAdapter()->getOrder($orderIdRequested);
         if (null === $order) {
             // this is not our order. access denied
             $viewData['error'] = true;
             $viewData['errorCode'] = 'orderNotFoundError';
+        } else {
+            if (null === $extranetUserId || false === $viewOrderDetailHandler->orderIdBelongsToUser($orderIdRequested, $extranetUserId)) {
+                // this is not our order. access denied
+                $viewData['error'] = true;
+                $viewData['errorCode'] = 'notMyOrderError';
+            }
         }
 
         $viewData['order'] = $order;
```

Comment:
This is (still) done - even in the case of error above?

Reply:
The order should be mapped in any case, and will contain null in case of an error (same as before the change).